contracts/src/v0.8/functions/dev/1_0_0/FunctionsClient.sol

```diff
@@ -0,0 +1,142 @@
+   * @param req The initialized Functions.Request
+   * @param subscriptionId The subscription ID
+   * @param callbackGasLimit gas limit for the fulfillment callback
+   * @return requestId The generated request ID
+   */
+  function _sendRequest(
+    Functions.Request memory req,
+    uint64 subscriptionId,
+    uint32 callbackGasLimit,
+    bytes32 donId
+  ) internal returns (bytes32) {
+    bytes memory requestData = Functions.encodeCBOR(req);
+    bytes32 requestId = _sendRequestBytes(requestData, subscriptionId, callbackGasLimit, donId);
+    return requestId;
+  }
+
+  /**
+   * @notice Sends a Chainlink Functions request to the stored oracle address
+   * @param data The initialized Functions request data
+   * @param subscriptionId The subscription ID
+   * @param callbackGasLimit gas limit for the fulfillment callback
+   * @return requestId The generated request ID
+   */
+  function _sendRequestBytes(
+    bytes memory data,
+    uint64 subscriptionId,
+    uint32 callbackGasLimit,
+    bytes32 donId
+  ) internal returns (bytes32) {
+    bytes32 requestId = s_router.sendRequest(
+      subscriptionId,
+      data,
+      Functions.REQUEST_DATA_VERSION,
+      callbackGasLimit,
+      donId
+    );
+    s_pendingRequests[requestId] = s_router.getRoute(donId);
+    emit RequestSent(requestId);
+    return requestId;
+  }
+
+  /**
+   * @notice User defined function to handle a response
+   * @param requestId The request ID, returned by sendRequest()
+   * @param response Aggregated response from the user code
+   * @param err Aggregated error from the user code or from the execution pipeline
+   * Either response or error parameter will be set, but never both
+   */
+  function fulfillRequest(bytes32 requestId, bytes memory response, bytes memory err) internal virtual;
+
+  /**
+   * @inheritdoc IFunctionsClient
+   */
+  function handleOracleFulfillment(
+    bytes32 requestId,
+    bytes memory response,
+    bytes memory err
+  ) external override recordChainlinkFulfillment(requestId) {
+    fulfillRequest(requestId, response, err);
+  }
+
+  /**
+   * @notice Sets the stored router address
+   * @param router The address of Functions router contract
+   */
+  function setRouter(address router) internal {
+    s_router = IFunctionsRouter(router);
+  }
+
+  /**
+   * @notice Gets the stored address of the router contract
+   * @return The address of the router contract
+   */
+  function getChainlinkFunctionsRouterAddress() internal view returns (address) {
+    return address(s_router);
+  }
+
+  /**
+   * @notice Allows for a request which was created on another contract to be fulfilled
+   * on this contract
+   * @param oracleAddress The address of the oracle contract that will fulfill the request
+   * @param requestId The request ID used for the response
+   */
+  function addExternalRequest(address oracleAddress, bytes32 requestId) internal notPendingRequest(requestId) {
```

Comment:
I am wondering if this function should be here.  We currently don't support a request from one contract triggering a fulfillment in another contract, so having it here might confuse users.

What is the reason it is here now?
I know I had something similar purely for testing purposes in our HH repo client contract to simulate a fulfillment, but once I create a proper e2e simulator tool, there should be no need for this.